Change installation documentation for Arch Linux

alexjorgef opened the pull request:
## Page

https://docs.soliditylang.org/en/latest/installing-solidity.html#linux-packages

## Description

According to [2d5142adfcc33148b9208083c09e21547a044abd](https://github.com/archlinux/svntogit-community/commit/2d5142adfcc33148b9208083c09e21547a044abd), `solidity` official package was removed. This changes Solidity's documentation to reference the two packages added in Arch User Repository (AUR), [solidity](https://aur.archlinux.org/packages/solidity) and [solidity-bin](https://aur.archlinux.org/packages/solidity-bin).

r0qs commented:
Hum...I wonder why it was removed, is that related to the [Archlinux git migration](https://archlinux.org/news/git-migration-announcement/)? Because now the `community` repository was merged into `extra`. So maybe `solidity` was supposed to be moved there as well but it was not?

cameel commented:
I doubt it was just forgotten. It was one of the Arch Linux devs who submitted it to AUR. I tried to find something on the mailing lists, but I see no mentions of it though.

In the end, packages get moved between AUR and official repos all the time. Maybe they just did not think maintaining it was worth it. After all, pretty much all frameworks just download binaries from solc-bin because you usually need a specific version. On a rolling release distro like Arch you only get the latest version packaged. I do use that package for the convenience of having a global `solc` binary always at hand, but I doubt it's seeing that much production use by people who actually develop contracts on Arch.

cameel commented on docs/installing-solidity.rst:
```diff
@@ -132,11 +132,7 @@ The nightly version can be installed using these commands:
-.. code-block:: bash
-
-    pacman -S solidity
+For example, Arch Linux has packages for the latest development version as AUR packages: `solidity` and `solidity-bin`.
```
```suggestion
For example, Arch Linux has packages for the latest development version as AUR packages: ``solidity`` and ``solidity-bin``.
```

cameel replied:
Well, technically Arch does not *have* these packages. These are just build scripts maintained by random users and cannot be trusted to the same level as ones actually provided by the distro. People using AUR should know that, but still, this may be worth a note since just mentioning them like this sounds like we're recommending them.

To be fair these scripts look perfectly fine *right now* (`solidity` is based on the original official script and `solidity-bin` just installs our binary from a Github release), but they can be changed at any time without any kind of vetting from the distro.

r0qs commented:
> The package was removed as part of the 23 spring cleanup, meaning it was orphaned on the `community` repo, and since no one wanted to maintain it, it was moved to the AUR. Hope this clarifies it +1
> 
> https://lists.archlinux.org/archives/list/arch-dev-public@lists.archlinux.org/thread/GEVNHHJLZUKAJB5CSIZWBU2UFRBV4232/ https://aur.archlinux.org/cgit/aur.git/commit/?h=solidity&id=3990c76bff3c033e9494b837916ec3b1ef8d1f0d

Thanks @FFY00, yes it does ;)

@alexjorgef thanks for your contribution, but I think this comment https://github.com/ethereum/solidity/pull/14281#discussion_r1211509191 should be addressed before we merge this PR. I.e. we need to be clear that the `solidity` package is not available on Archlinux but it is possible to install it using the community-driven repository (maybe adding a link to the AUR documentation), and users should be aware of the risks in doing it.

nagrarohit reviewed (commented):
@alexjorgef  hey Bro. can you please provide the link for aur packages of arch linux

alexjorgef committed:
Change installation documentation for Arch Linux


Correct inline code with double backquotes


Add more information

Add external links to packages;
Add a note to inform that packages don't have official support.

alexjorgef commented:
> > The package was removed as part of the 23 spring cleanup, meaning it was orphaned on the `community` repo, and since no one wanted to maintain it, it was moved to the AUR. Hope this clarifies it +1
> > https://lists.archlinux.org/archives/list/arch-dev-public@lists.archlinux.org/thread/GEVNHHJLZUKAJB5CSIZWBU2UFRBV4232/ https://aur.archlinux.org/cgit/aur.git/commit/?h=solidity&id=3990c76bff3c033e9494b837916ec3b1ef8d1f0d
> 
> Thanks @FFY00, yes it does ;)
> 
> @alexjorgef thanks for your contribution, but I think this comment [#14281 (comment)](https://github.com/ethereum/solidity/pull/14281#discussion_r1211509191) should be addressed before we merge this PR. I.e. we need to be clear that the `solidity` package is not available on Archlinux but it is possible to install it using the community-driven repository (maybe adding a link to the AUR documentation), and users should be aware of the risks in doing it.

Thanks @FFY00 for clarifying the information. It appears that Arch is making enhancements to their delivery system, enabling a stronger and more automated approach for users to contribute. Hopefully, we will see official package reinstated.

r0qs commented:
LGTM. @alexjorgef could you please squash the commits and rebase it? Then I think we can merge it ;)

alexjorgef commented:
> LGTM. @alexjorgef could you please squash the commits and rebase it? Then I think we can merge it ;)

@r0qs Done, I apologize for the delay :smile:
